### PR TITLE
[Merged by Bors] - Migrate integration tests

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- Moved tests to operator repo
- Adapted to [ADR018](https://docs.stackable.tech/home/contributor/adr/ADR018-product_image_versioning.html)

Integration tests removed in https://github.com/stackabletech/integration-tests/pull/249

## Review Checklist

- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by backstreetkiwi — APPROVED** [review]
test ran successfully: https://ci.stackable.tech/view/02%20Component%20Tests%20(flex)/job/Trino%20Operator%20Integration%20Tests%20(flex)/15/

**Comment by maltesander** [comment]
> @maltesander The Trino test was successful. Do you need another approval from an operator dev or is my approval sufficient?

There were some changes to the code (not just the migration of the tests), so it would make sense to wait.

**Review by razvan — APPROVED** [review]
Works but I'm not sure if "0.0.{}" is the best choice.

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -739,22 +750,27 @@ fn build_rolegroup_service(
...
 }
 
-/// Returns the "real" Trino version for docker images e.g. 377
-pub fn trino_version_trim(trino: &TrinoCluster) -> Result<&str> {
-    let spec: &TrinoClusterSpec = &trino.spec;
-    spec.version
-        .as_deref()
-        .and_then(|v| v.split('.').collect::<Vec<_>>().last().cloned())
-        .context(ObjectHasNoVersionSnafu)
+/// Returns our semver representation for product config e.g. 0.0.377
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs]: done

**Review by razvan — APPROVED** [review]
lgtm

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**